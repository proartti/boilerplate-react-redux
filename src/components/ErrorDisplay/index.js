/* @flow */

import React from 'react';

import styles from './styles.scss';

type Props = { error: Object };

export default ({ error }: Props) => (
  <div className={styles.Error}>
    <p>Oops! {error.message}</p>
  </div>
);
