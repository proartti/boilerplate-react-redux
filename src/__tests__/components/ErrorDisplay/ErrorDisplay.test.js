import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

import ErrorDisplay from '../../../components/ErrorDisplay/';

describe('<ErrorDisplay />', () => {
  test('renders', () => {
    const mockData = { message: 'Error!' };
    const tree = renderer
      .create(
        <StaticRouter context={{}}>
          <ErrorDisplay error={mockData} />
        </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
