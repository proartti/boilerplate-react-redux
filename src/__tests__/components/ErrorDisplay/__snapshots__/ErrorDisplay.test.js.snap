// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorDisplay /> renders 1`] = `
<div
  className="Error"
>
  <p>
    Oops! 
    Error!
  </p>
</div>
`;
